uniffi_core/src/ffi/rustfuture/mod.rs

```diff
@@ -28,10 +31,23 @@ pub enum RustFuturePoll {
 ///
 /// The Rust side of things calls this when the foreign side should call [rust_future_poll] again
 /// to continue progress on the future.
-pub type RustFutureContinuationCallback = extern "C" fn(callback_data: *const (), RustFuturePoll);
+///
+/// WARNING: the call to [rust_future_poll] must be scheduled to happen soon after the callback is
+/// called, but not inside the callback itself.  If [rust_future_poll] is called inside the
+/// callback, some futures will deadlock and our scheduler code might as well.
```

Comment:
This isn't really related to the PR, but it's something I noticed when trying to refactor some things.